metaflow/plugins/aws/aws_utils.py

```diff
@@ -101,4 +102,17 @@ def compute_resource_attributes(decos, compute_deco, resource_defaults):
         if compute_deco.attributes.get(k) is not None:
             result[k] = str(compute_deco.attributes[k] or "0")
 
+    if "cpu" in result and MAX_CPU_PER_TASK:
+        if float(result["cpu"]) > float(MAX_CPU_PER_TASK):
+            raise MetaflowException(
+                "Step %s requires %s CPU units, but you cannot use more than %s per step in this environment"
+                % (step_name, result["cpu"], MAX_CPU_PER_TASK)
+            )
+    if "memory" in result and MAX_MEMORY_PER_TASK:
+        if float(result["memory"]) > float(MAX_MEMORY_PER_TASK):
+            raise MetaflowException(
+                "Step %s requires %s CPU units, but you cannot use more than %s per step in this environment"
+                % (step_name, result["memory"], MAX_MEMORY_PER_TASK)
+            )
+
```

Comment:
I wonder if a more flexible approach here would be to have this block for all attributes in the `resource_defaults` or something like that. This seems very specific to just two elements for some reason.

Reply:
Yes, its mostly because I thought I may generate more user-friendly error messages if I don't start with a generic approach, but looking at it they're not that different. Then again, can always make it more generic later